crates/eframe/src/native/glow_integration.rs

```diff
@@ -829,6 +829,18 @@ impl GlowWinitRunning {
                 }
             }
 
+            winit::event::WindowEvent::Destroyed => {
+                log::debug!(
+                    "Received WindowEvent::Destroyed for viewport {:?}",
+                    viewport_id
+                );
+                if viewport_id == Some(ViewportId::ROOT) {
+                    return EventResult::Exit;
+                } else {
+                    return EventResult::Wait;
+                }
+            }
+
```

Comment:
You probably need to add the same code to `wgpu_integration.rs`.

We just merged the update to latest `winit` (https://github.com/emilk/egui/pull/4849), so please merge that into this branch and then test this PR with:

* `cargo r -p test_viewports`
* `cargo r -p test_viewports -F wgpu`
* `cargo r -p serial_windows`

Reply:
will do, note that I found this https://github.com/emilk/egui/issues/4894

Reply:
@emilk Am I reading this correctly. That clicking the close box on a child viewport does nothing unless I actually have code that checks for CloseRequested? And that you have to set the vp as not visible in order to destroy it . Thats not at all obvious. (Was trying to work out why the bug reporters app was not closing the child VP after I fixed the bugs I could find)

Reply:
@emilk I added to wgpu the equivalent release of the drawing surface, in wgpu not releasing this resulted in the recreate of the vp leaving the original in place (rather than crashing as in glow).

Tested on windows and mac, 

Note the wgpu has very strange defferred vp close behavior (also meaning that I did not need to add any close logic to wpgpu_integration). Because there is no close option only the option to no longer request its presence there is some very odd stuff happening. I worried that it was my bad but the current master shows the same thing. After CloseRequest is received you have to stop drawing secondary vp in update, BUT, this means you need another main window draw before the window disappears. I filed a bug on this #4894 . I tried to add the necessary code to the original reports code but could not - I thought it was simply a matter of doing request_repaint but I could not make that work. And the fact that the test_viewports app shows the same bug shows that its not simple.

I think it would be really ice to have an explicit vp.close call (even though that goes again the immediate model VPs are really not transient, the OS knows them (unlike stuff drawn on the gl surfaces). I will investigate and make a separate PR

Also I have not tested on linux becasue I cannot make it build on linux, it complains about a lot of pkg errors.

I will do a separate PR to add vp calls the need window recreation to test_viewports.

In summary - I think this PR is complete, apart from not being tested on Linux.

Reply:
Thanks!